os/tools/check_output_size.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python
+###########################################################################
+#
+# Copyright 2021 Samsung Electronics All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+#
+###########################################################################
+
+# This script is for checking kernel partition is large enough to accommodate kernel binary.
+# For this, get the kernel partition size and kernel binary size,
+# and check the kernel binary size is smaller than kernel partition size.
+
+import os
+import sys
+import string
+
+cfg_file = os.path.dirname(__file__) + '/../.config'
+
+def get_config_value(file_name, config):
+    with open(file_name, 'r+') as f:
+        lines = f.readlines()
+	value = 'None'
+	for line in lines:
+		if config in line:
+			value = (line.split("=")[1])
+			break		
+	return value;
+
+PARTITION_SIZE_LIST = get_config_value(cfg_file, "CONFIG_FLASH_PART_SIZE=")
+PARTITION_NAME_LIST = get_config_value(cfg_file, "CONFIG_FLASH_PART_NAME=")
+
+if PARTITION_SIZE_LIST == 'None' :
+	sys.exit(0)
+
+NAME_LIST = PARTITION_NAME_LIST.split(",")
+SIZE_LIST = PARTITION_SIZE_LIST.split(",")
+
+# Find Kernel Partition Index
+KERNEL_IDX=0
+for name in NAME_LIST :
+	name = name.replace('"', '')
+	if name == "kernel" or name == "os" :
+		break
+	else :
+		KERNEL_IDX += 1
+
+KERNEL_PARTITION_SIZE=SIZE_LIST[KERNEL_IDX]
+
+# Check the board type. Because kernel binary name is different based on board type.
+BOARD_TYPE = get_config_value(cfg_file, "CONFIG_ARCH_BOARD=")
+BOARD_TYPE = BOARD_TYPE.replace('"', '')
+BOARD_TYPE = BOARD_TYPE.rstrip("\n")
+
+if BOARD_TYPE == "rtl8721csm" :
+	# If the board is rtl8721csm, the kernel binary name is km0_km4_image2.bin
+	output_path = os.path.dirname(__file__) + '/../../build/output/bin/km0_km4_image2.bin'
+else :
+	# Other cases, the kernel binary name is tinyara.bin
+	output_path = os.path.dirname(__file__) + '/../../build/output/bin/tinyara.bin'
+
+# In partition list, its size is KB, so calculate the size as KB.
+KERNEL_BINARY_SIZE=os.path.getsize(output_path) / 1024
+
+# Compare the kernel partition size and kernel binary size
+if KERNEL_PARTITION_SIZE < KERNEL_BINARY_SIZE:
+	print("!!!!!!!! ERROR !!!!!!!")
+	print("Kernel Binary size(" + str(KERNEL_BINARY_SIZE) + ") is greater than kernel partition size(" + str(KERNEL_PARTITION_SIZE) + ").")
```

Comment:
You can change it like below.
`print("Kernel Binary size %d is greater ... %d" %(KERNEL_BINARY_SIZE, KERNEL_PARTITION_SIZE))`

Reply:
I will keep as of now.